### DISCUSSION TITLE
Is there support for nanoseconds (timestamp)?

### DISCUSSION THREAD

**Discussion by straxess** [logfellow/logstash-logback-encoder#799] [Q&A] [answered]
Is there support for nanoseconds?

**Comment by philsttr** [comment, marked as answer]
Not currently.

logback 1.2.x does not support nanoseconds.  Therefore logstash-logback-encoder cannot support nanoseconds at the moment.

logback 1.3.x (not released yet) added support for nanoseconds in [LOGBACK-1374](https://jira.qos.ch/browse/LOGBACK-1374).  Whever logback 1.3.x is released, logstash-logback-encoder will need to be modified to support nanoseconds.

**Reply by philsttr** [reply]
nanosecond support added in #833 (requires logback 1.3.0+)

**Reply by brenuart** [reply]
... and logstash-Logback-encoder 7.3

**Comment by wsargent** [comment]
You can always add your own field using `instant.toString` or wrap your appender with https://tersesystems.github.io/terse-logback/1.0.0/guide/relativens/ to add a nanosecond element to your logging event.
